### PR TITLE
[NOD-1497] Add additional methods to consensus external API

### PR DESCRIPTION
**Pull request by stasatdaglabs**

### CODE REVIEW

File: domain/consensus/consensus.go [hunk 1]
```diff
@@ -10,6 +10,17 @@ type Consensus interface {
 	BuildBlock(coinbaseData *externalapi.DomainCoinbaseData, transactions []*externalapi.DomainTransaction) (*externalapi.DomainBlock, error)
 	ValidateAndInsertBlock(block *externalapi.DomainBlock) error
 	ValidateTransactionAndPopulateWithConsensusData(transaction *externalapi.DomainTransaction) error
+
+	GetBlock(blockHash *externalapi.DomainHash) (*externalapi.DomainBlock, error)
```
Comment (someone235) [inline domain/consensus/consensus.go:14]: Please add SetPruningUTXOSet(hash Hash, serializedUTXOSet []byte)

File: domain/consensus/consensus.go [hunk 2]
```diff
@@ -10,6 +10,17 @@ type Consensus interface {
...
+	GetBlockHeader(blockHash *externalapi.DomainHash) (*externalapi.DomainBlockHeader, error)
+	GetBlockInfo(blockHash *externalapi.DomainHash) (*externalapi.BlockInfo, error)
+
+	GetHashesBetween(lowHigh, highHash *externalapi.DomainHash) ([]*externalapi.DomainHash, error)
+	GetHashesAbovePruningPoint(highHash *externalapi.DomainHash) ([]*externalapi.DomainHash, error)
```
Comment (someone235) [inline domain/consensus/consensus.go:19]: If the name is entirely wrong, please add "TBD" or something like this

File: domain/consensus/consensus.go [hunk 3]
```diff
@@ -10,6 +10,17 @@ type Consensus interface {
...
+	GetPruningPointUTXOSet() ([]byte, error)
+	GetSelectedParent() (*externalapi.DomainBlock, error)
```
Comment (someone235) [inline domain/consensus/consensus.go:21]: Rename to GetSelectedTip, or GetVirtualSelectedParent

File: domain/consensus/consensus.go [hunk 4]
```diff
@@ -10,6 +10,17 @@ type Consensus interface {
 	BuildBlock(coinbaseData *externalapi.DomainCoinbaseData, transactions []*externalapi.DomainTransaction) (*externalapi.DomainBlock, error)
 	ValidateAndInsertBlock(block *externalapi.DomainBlock) error
 	ValidateTransactionAndPopulateWithConsensusData(transaction *externalapi.DomainTransaction) error
+
```
Comment (someone235) [inline domain/consensus/consensus.go:13]: What about SelectedTipBlueScore?

**Review by someone235 — CHANGES_REQUESTED** [review]

File: domain/consensus/consensus.go (hunk 4 above)
Comment (stasatdaglabs) [reply domain/consensus/consensus.go:13]: The blue score is hidden in the coinbase of the selected tip block.

**Review by someone235 — APPROVED** [review]

**Pull request merged**